rpc-api/cmd/notify-rpc-api/grpc.go

```diff
@@ -0,0 +1,447 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/govau/notify/rpc-api/notify"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+	jose "gopkg.in/square/go-jose.v2"
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+type Client struct {
+	http.Client
+
+	BaseURL     *url.URL
+	ServiceID   string
+	API_Key     string
+	RouteSecret string
+}
+
+func CreateJWT(clientID, secret string) (string, error) {
+	key := jose.SigningKey{Algorithm: jose.HS256, Key: []byte(secret)}
+	sig, err := jose.NewSigner(key, (&jose.SignerOptions{}).WithType("JWT"))
+	if err != nil {
+		return "", err
+	}
+
+	cl := jwt.Claims{
+		Issuer:   clientID,
+		IssuedAt: jwt.NewNumericDate(time.Now()),
+	}
+
+	return jwt.Signed(sig).Claims(cl).CompactSerialize()
+	return jwt.Signed(sig).Claims(cl).FullSerialize()
+}
+
+func (c Client) Do(req *http.Request) (*http.Response, error) {
+	token, err := CreateJWT(c.ServiceID, c.API_Key)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Add("Content-type", "application/json")
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", token))
+	req.Header.Add("X-Custom-Forwarder", "")
+	req.Header.Add("User-agent", "NOTIFY-API-PYTHON-CLIENT/4.9.0")
+
+	req.URL.Host = ""
+	req.URL.Scheme = ""
+	req.URL = c.BaseURL.ResolveReference(req.URL)
+
+	return c.Client.Do(req)
+}
+
+func (c Client) Get(path string) (*http.Response, error) {
+	req, err := http.NewRequest("GET", path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode > 300 {
+		return resp, errors.New("bad status code")
```

Comment:
Should we do `return resp, fmt.Errorf("bad status code %d", resp.StatusCode)` so that it's a bit easier to debug issues?